src/jsrs_parser.cc

```diff
@@ -74,28 +76,30 @@ static constexpr Local<Value> (*kParseFunctions[])(Isolate*,
 };
 
 Local<Value> Parse(Isolate* isolate, const String::Utf8Value& in) {
-  size_t size;
-  const char* to_parse =
-      internal::PrepareString(isolate, *in, in.length(), &size);
-  if (!to_parse) {
-    return Undefined(isolate);
-  }
+  const char* str = *in;
+  const size_t length = in.length();
+  const char* end = str + length;
 
   Type type;
-  if (!GetType(to_parse, to_parse + size, &type)) {
+
+  size_t start_pos = internal::SkipToNextToken(str, end);
```

Comment:
Here it is not the case, because `SkipToNextToken` returns `size_t`, which is fine (there is no pointer subtraction to get result in this function), but generally you are right and there are other occurrences where it can be important. I guess, I will still review and check for any issues that may be caused by this.